circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/Replica.java

```diff
@@ -426,4 +436,25 @@ public TableAndStatistics getTableAndStatistics(TableReplication tableReplicatio
     return super.getTableAndStatistics(tableReplication.getReplicaDatabaseName(),
         tableReplication.getReplicaTableName());
   }
+
+  private void dropReplicaTable(CloseableMetaStoreClient client, String replicaDatabaseName, String replicaTableName) {
```

Comment:
Would recommend moving this method to the `DropTableService`. Also, on a full overwrite I imagine we will want to delete the current data (right?), but if we're using Beekeeper drop table events will be ignored (unless we add the param to allow drop table events). But even if we do this, we won't want to schedule the table path because then Beekeeper will delete any new data that is added after the delete - Beekeeper doesn't yet go to the Metastore to only schedule the current partitions for deletion on a table drop.

We probably want to do an alter all the individual partitions to make sure they're deleted and not the new data? Not sure, is that possible?

Reply:
Spoke with the team, we think that this is true for unpartitioned tables, but for partitioned tables the table path doesn't include the event id, the partition paths do instead. So we can't do a drop without first making sure that Beekeeper isn't listening otherwise we'll lose all the new data too.